packages/fx-core/src/common/samples.ts

```diff
@@ -67,56 +70,133 @@ class SampleProvider {
+        } else if (version.includes("rc")) {
+          // if there is a breaking change, the tag is not used by any stable version.
+          return TeamsSampleConfigTag;
+        } else {
+          // stable version uses the head of branch defined by feature flag when available
+          return TeamsSampleConfigTag;
+        }
+      } else {
+        // Office Samples
+        if (version.includes("alpha")) {
+          return "dev";
+        } else if (version.includes("beta")) {
+          return SampleConfigBranchForPrerelease;
+        } else if (version.includes("rc")) {
+          return OfficeSampleConfigTag;
+        } else {
+          return OfficeSampleConfigTag;
+        }
+        // return "dev";
+      }
+    };
     const version: string = packageJson.version;
-    const configBranchInEnv = process.env[FeatureFlagName.SampleConfigBranch];
+    const configBranchInEnv =
+      process.env[
+        configRepo === TeamsSampleConfigRepo
+          ? FeatureFlagName.TeamsSampleConfigBranch
+          : FeatureFlagName.OfficeSampleConfigBranch
+      ];
     let samplesConfig: SampleConfigType | undefined;
-    let ref = SampleConfigTag;
-
-    // Set default value for branchOrTag
-    if (version.includes("alpha")) {
-      // daily build version always use 'dev' branch
-      ref = "dev";
-    } else if (version.includes("beta")) {
-      // prerelease build version always use branch head for prerelease.
-      ref = SampleConfigBranchForPrerelease;
-    } else if (version.includes("rc")) {
-      // if there is a breaking change, the tag is not used by any stable version.
-      ref = SampleConfigTag;
-    } else {
-      // stable version uses the head of branch defined by feature flag when available
-      ref = SampleConfigTag;
-    }
-
+    let ref = getRef(configRepo, version);
     // Set branchOrTag value if branch in env is valid
     if (configBranchInEnv) {
       try {
-        const data = await this.fetchRawFileContent(configBranchInEnv);
+        const data = await this.fetchRawFileContent(
+          SampleConfigOwner,
+          configRepo,
+          configBranchInEnv,
+          configFile
+        );
         ref = configBranchInEnv;
         samplesConfig = data as SampleConfigType;
       } catch (e: unknown) {}
     }
 
     if (samplesConfig === undefined) {
-      samplesConfig = (await this.fetchRawFileContent(ref)) as SampleConfigType;
+      samplesConfig = (await this.fetchRawFileContent(
+        SampleConfigOwner,
+        configRepo,
+        ref,
+        configFile
+      )) as SampleConfigType;
     }
 
     return { samplesConfig, ref };
   }
 
   @hooks([ErrorContextMW({ component: "SampleProvider" })])
-  private parseOnlineSampleConfig(samplesConfig: SampleConfigType, ref: string): SampleCollection {
+  private parseOnlineSampleConfig(
+    samplesOnwer: string,
+    samplesRepo: string,
+    samplesConfig: SampleConfigType,
+    ref: string
+  ): Promise<SampleCollection> {
```

Comment:
Why does this function return `Promise`?

Reply:
Because the ErrorContextMW middleware transfer the parseOnlineSampleConfig into a async function. I need to convert this function into async function to get the right result.

![image](https://github.com/OfficeDev/TeamsFx/assets/105693302/1c5fafcc-05ef-4056-95e6-f2da80b2d761)